src/Trinity.Core/Trinity/Configuration/TrinityConfig.IO.cs

```diff
@@ -112,102 +112,124 @@ private static ConfigurationInstance CreateConfigurationEntryTuple(Type type)
         /// </summary>
         public static void SaveConfig()
         {
-            SaveConfig(DefaultConfigFile);
+            try
+            {
+                SaveConfig(DefaultConfigFile);
```

Comment:
public interface should not eat the exception. add a statement `throw;` in the catch block to re-throw it.
Also, revise the log message. Tell people why it fails.